backend/modules/professor/routers/professor_cohorts.py

```
@@ -0,0 +1,379 @@
+    """Recalculate time_spent for all student instances in the professor's cohorts."""
+    try:
+        result = service.refresh_assigned_simulations(current_user.id)
+        return {"status": "ok", **result}
+    except Exception as e:
+        logger.error(f"Error in refresh_assigned_simulations: {str(e)}", exc_info=True)
+        raise HTTPException(status_code=500, detail="Failed to refresh assignments")
+
+
+@router.get("/admin/all", response_model=List[CohortListResponse])
+async def get_all_cohorts_admin(
+    current_user: User = Depends(require_professor),
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)